ev3dev/core.py

```diff
@@ -2038,7 +2049,17 @@ def mode(self):
 
     @mode.setter
     def mode(self, value):
+        """
+        This will be called often so we cache the current mode in
+        self._mode_value to avoid some file IO.
+        """
+
+        # If 'mode' is already set to 'value' there is nothing to do.
+        if value == self._mode_value:
+            return
+
         self._mode = self.set_attr_string(self._mode, 'mode', value)
+        self._mode_value = value
 
```

Comment:
Don't you need to update the getter as well?

Reply:
True I could just have it return self._mode_value